### PR TITLE
Added keymode to constructor params to avoid calling _deriveDukptSess…

### PR DESCRIPTION
**Pull request by awebdev**
Added keymode to constructor params to avoid calling _deriveDukptSessionKey twice to use non-default modes

### CODE REVIEW

**Comment by awebdev** [comment]
@dpjayasekara can you check this out.

It helps passing key mode to constructor, supports running node-dukpt tests on node v4.8.3

**Comment by dpjayasekara** [comment]
@awebdev Will do that. Thanks for the contribution.

File: lib/dukpt.lib.js [hunk 1]
```diff
@@ -2,10 +2,10 @@ const aesjs = require('aes-js');
 const DataOperations = require('./data.lib');
 
 class Dukpt {
-    constructor(bdk, ksn) {
+    constructor(bdk, ksn, keymode = 'datakey') {
         this.bdk = bdk;
         this.ksn = ksn;
-        this._sessionKey = this._deriveDukptSessionKey();
+        this._sessionKey = this._deriveDukptSessionKey(datakey);
```
Comment (dpjayasekara) [inline lib/dukpt.lib.js]: datakey is not defined anywhere

**Comment by dpjayasekara** [comment]
@awebdev Please kindly note that there are multiple issues in this pull request which I have commented on. As other key modes are not supported yet, providing keymode in the constructor is discouraged. Also, since this is merely a minor improvement, will not be merged right now.

File: lib/dukpt.lib.js (hunk 1 above)
Comment (awebdev) [inline lib/dukpt.lib.js]: did you look at just the first commit? `datakey` was renamed to `keyMode` in my last commit.

**Comment by awebdev** [comment]
@dpjayasekara I do agree these are minor changes but calling `_deriveDukptSessionKey` twice to decrypt with `pinkey` does not make any sense.

**Comment by dpjayasekara** [comment]
Yes, these minor issues will be addressed and some performance improvements will be implemented soon in a new release. Please kindly note that this pull request contains multiple issues as I commented due to which I cannot merge this.

File: lib/dukpt.lib.js (hunk 1 above)
Comment (dpjayasekara) [inline lib/dukpt.lib.js]: noted

**Pull request closed without merging**